eppo_client/sharders.py

```diff
@@ -0,0 +1,30 @@
+from abc import ABC, abstractmethod
+from typing import Dict
+import hashlib
+
+
+class Sharder(ABC):
+    @abstractmethod
+    def get_shard(self, input: str, total_shards: int) -> int: ...
+
+
+class MD5Sharder(Sharder):
+    def get_shard(self, input: str, total_shards: int) -> int:
+        hash_output = hashlib.md5(input.encode("utf-8")).hexdigest()
+        # get the first 4 bytes of the md5 hex string and parse it using base 16
+        # (8 hex characters represent 4 bytes, e.g. 0xffffffff represents the max 4-byte integer)
+        int_from_hash = int(hash_output[0:8], 16)
+        return int_from_hash % total_shards
+
+
+class DeterministicSharder(Sharder):
+    """
+    Deterministic sharding based on a look-up table
+    to simplify writing tests
+    """
+
+    def __init__(self, lookup: Dict[str, int]):
+        self.lookup = lookup
+
+    def get_shard(self, input: str, total_shards: int) -> int:
+        return self.lookup.get(input, 0)
```

Comment:
please add unit tests for this

Reply:
The deterministic sharder is there only for unit testing purposes

Reply:
Will add the same tests as I just did to the JS commons lib: https://github.com/Eppo-exp/js-client-sdk-common/pull/52/files#diff-ea83fb75e9f26161475fd7e79c9bd4eb4292dc61b8b02161d43baa036f5d6140